app/controllers/jobseekers/job_applications/details_controller.rb

```diff
@@ -0,0 +1,78 @@
+class Jobseekers::JobApplications::DetailsController < Jobseekers::ApplicationController
+  helper_method :back_link_path, :build_step, :detail, :form, :job_application
+
+  def create
+    if form.valid?
+      job_application.job_application_details.create(details_type: build_step, data: detail_params)
+      if params[:commit] == t("jobseekers.job_applications.details.form.#{build_step}.add_another")
+        redirect_to new_jobseekers_job_application_build_detail_path(job_application, build_step)
+      else
+        redirect_to back_link_path
+      end
+    else
+      render :new
+    end
+  end
+
+  def update
+    if form.valid?
+      detail.update(data: detail.data.merge(detail_params))
+      redirect_to back_link_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    detail.destroy
+    redirect_to back_link_path, success: t("messages.jobseekers.job_applications.#{build_step}.deleted")
+  end
+
+  private
+
+  def back_link_path
+    @back_link_path ||= jobseekers_job_application_build_path(job_application, build_step)
+  end
+
+  def build_step
+    @build_step ||= params[:build_id]
```

Comment:
I can't find a reference to build_id in latest master or this branch `*scratches head*`

Reply:
`bundle exec rails routes` (it comes from the wicked gem and refers to the current step)